iree/hal/cuda/graph_command_buffer.c

```diff
@@ -286,16 +374,16 @@ static iree_status_t iree_hal_cuda_graph_command_buffer_fill_buffer(
       .height = 1,
       .value = dword_pattern,
   };
-  // Serialize all the nodes for now.
-  CUgraphNode dep[] = {command_buffer->last_node};
-  size_t numNode = command_buffer->last_node ? 1 : 0;
+  CUgraphNode dep[CUDA_MAX_LEAF_NODES];
+  size_t numNode = iree_cuda_get_dep_nodes(command_buffer, &dep);
+  CUgraphNode node;
```

Comment:
nit: always good to initialize things in C, even if they are overwritten
```suggestion
  CUgraphNode node = NULL;
```
(as a general rule, always initializing everything is good - knowing when not to requires more global knowledge that's easy to break or miss)